### PR TITLE
use scalafmt to format sources (instead of scalariform)

### PR DESCRIPTION
**Pull request by SethTisue**
wdyt @Philippus ?

note that I'm not enforcing this in CI anymore; that seems annoying to me

but Scala Steward will reformat sources when it bumps the Scalafmt version. that seems sufficient to me

Scalariform isn't maintained anymore, and will surely never make the jump to Scala 3

### CODE REVIEW

**Comment by Phillipus** [comment]
> wdyt @Phillipus?

I think you pinged the wrong person.

**Review by Philippus — COMMENTED** [review]
I think it's good to phase out scalariform.

File: .scalafmt.conf [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+maxColumn = 100
+newlines.source = keep
+newlines.alwaysBeforeElseAfterCurlyIf = true
+indentOperator.excludeRegex = "^$"
+danglingParentheses.defnSite = false
+danglingParentheses.callSite = false
+rewrite.trailingCommas.style = keep
+binPack.literalsExclude = []
+binPack.literalsIncludeSimpleExpr = true
+docstrings.style = Asterisk
```
Comment (Philippus) [inline .scalafmt.conf:12]: Maybe just go with the default settings? Not sure where these are coming from. :)

**Comment by SethTisue** [comment]
Oh, I initially just copied my personal stylesheet without thinking about it too much.

Just now, I pushed a second commit that takes a different approach: mostly stick to defaults, but include a few things that reflect style choices in the existing codebase, thus keeping the diff smaller.

Also I just really disagree with 80, I think it's too narrow, and according to https://twitter.com/alexelcu/status/1481292324026130432 a lot of people agree.

**Pull request merged**